src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -943,6 +944,19 @@ def register_strategies(cls, strategy_registry: Dict) -> None:
             offload_optimizer_device="nvme",
         )
 
+    def batch_to_device(self, batch: Any, *args: Any, **kwargs: Any) -> Any:
+        def fp_to_half(tensor: Tensor) -> Tensor:
+            if torch.is_floating_point(tensor):
+                if self.precision_plugin.precision == PrecisionType.HALF:
+                    return tensor.half()
+                elif self.precision_plugin.precision == PrecisionType.BFLOAT:
+                    return tensor.bfloat16()
+
+            return tensor
+
+        batch = apply_to_collection(batch, Tensor, function=fp_to_half)
```

Comment:
should we move this to common utils? This is almost the same as with IPU and we should be able to exactly reuse that function as for IPU the precisiontype should always be HALF

Reply:
done!